### DISCUSSION TITLE
SPA support?

### DISCUSSION THREAD

**Discussion by DavidBernal** [babashka/nbb#117] [Q&A]
Hi, I found this project very interesting because is a easy way to start on clojure world with all JS ecosystem. 

I don't see way to use nbb to create a SPA with react for example. Is possible do it compiling cljs to js with nbb?

**Comment by borkdude** [comment]
You are probably looking for https://babashka.org/scittle/ ?

**Comment by borkdude** [comment]
nbb / scittle, etc are not compiling CLJS but interpreting it. If you want to compile your project, then I can recommend shadow-cljs.

**Comment by Sleepful** [comment]
I think it would be very interesting if NBB could be made to work with metaframeworks, but it is probably incompatible to do so due to the nature of the interpreter, whilst metaframeworks rely on a build step. Metaframeworks refers to things like NextJS and SolidStart, which rely not only on valid JS, but also on file-based structure.

**Comment by borkdude** [comment]
You can also look at squint or cherry nowadays.

https://github.com/squint-cljs/squint
